compiler-plugin/src/main/kotlin/arrow/meta/autofold/AutoFoldPlugin.kt

```diff
@@ -0,0 +1,76 @@
+package arrow.meta.autofold
+
+import arrow.meta.extensions.ExtensionPhase
+import arrow.meta.extensions.MetaComponentRegistrar
+import arrow.meta.higherkind.arity
+import arrow.meta.higherkind.isKinded
+import arrow.meta.kt.SealedSubclass
+import arrow.meta.kt.renderSuperTypes
+import arrow.meta.kt.sealedSubclasses
+import arrow.meta.qq.classOrObject
+import org.jetbrains.kotlin.cli.common.messages.CompilerMessageLocation
+import org.jetbrains.kotlin.cli.common.messages.CompilerMessageSeverity
+import org.jetbrains.kotlin.cli.common.messages.MessageCollector
+import org.jetbrains.kotlin.psi.KtClass
+import org.jetbrains.kotlin.psi.KtNamedFunction
+import org.jetbrains.kotlin.psi.KtParameterList
+import org.jetbrains.kotlin.psi.KtValueArgumentList
+import org.jetbrains.kotlin.psi.psiUtil.findFunctionByName
+import org.jetbrains.kotlin.utils.addToStdlib.safeAs
+
+val MetaComponentRegistrar.autoFold: List<ExtensionPhase>
+  get() =
+    meta(
+      classOrObject(::isAutoFoldable) { c ->
+        println("Processing Autofoldable: ${c.name} with superTypes:${c.renderSuperTypes()}")
```

Comment:
```suggestion
```

Reply:
```suggestion
        println("Create Autofold for: ${c.name}")
```
should it be just this, because